src/adaptors/morpho/index.js

```diff
@@ -49,37 +75,40 @@ const main = async () => {
         +marketFromGraph.p2pData.p2pSupplyIndex) /
       `1e${18 + marketFromGraph.token.decimals}`;
     const totalSupply = totalSupplyOnPool + totalSupplyP2P;
-    const totalBorrowOnPool =
-      (+marketFromGraph.metrics.borrowBalanceOnPool *
-        +marketFromGraph.reserveData.borrowPoolIndex) /
-      `1e${18 + marketFromGraph.token.decimals}`;
-    const totalBorrowP2P =
-      (+marketFromGraph.metrics.borrowBalanceInP2P *
-        +marketFromGraph.p2pData.p2pBorrowIndex) /
-      `1e${18 + marketFromGraph.token.decimals}`;
-    const totalBorrow = totalBorrowOnPool + totalBorrowP2P;
     const tvlUsd =
-      (totalSupply - totalBorrow) *
+      totalSupply *
       (marketFromGraph.reserveData.usd /
         `1e${18 * 2 - marketFromGraph.token.decimals}`);
 
     const poolSupplyRate = +marketFromGraph.reserveData.supplyPoolRate;
     const poolBorrowRate = +marketFromGraph.reserveData.borrowPoolRate;
 
     const p2pIndexCursor = +marketFromGraph.p2pIndexCursor / 1e4;
-    const poolSupplyAPY = rateToAPY(poolSupplyRate / 1e18);
-    const poolBorrowAPY = rateToAPY(poolBorrowRate / 1e18);
+    const poolSupplyAPY = rateToAPY((poolSupplyRate / 1e18) * BLOCKS_PER_DAY);
+    const poolBorrowAPY = rateToAPY((poolBorrowRate / 1e18) * BLOCKS_PER_DAY);
+
     const spread = poolBorrowAPY - poolSupplyAPY;
     const p2pSupplyAPY = poolSupplyAPY + spread * p2pIndexCursor;
     const avgSupplyAPY =
-      (totalSupplyOnPool * poolSupplyAPY + totalSupplyP2P * p2pSupplyAPY) /
-      totalSupply;
+      totalSupply === 0
+        ? 0
+        : (totalSupplyOnPool * poolSupplyAPY + totalSupplyP2P * p2pSupplyAPY) /
+          totalSupply;
+    const compAPY = computeCompRewardsAPY(marketFromGraph, compPrice);
+    const avgCompSupplyAPY =
+      totalSupply === 0 ? 0 : (compAPY * totalSupplyOnPool) / totalSupply; // Morpho redistributes comp rewards to users on Pool
+
+    const morphoRewards = 0; // MORPHO token is not transferable for now,
```

Comment:
maybe better to check if it is transferable using rpc? so that you don't have to redo it again later